dev/traffic_router/Dockerfile

```diff
@@ -15,12 +15,17 @@ FROM alpine:latest AS trafficrouter-dev
 
 ENV TC=/root/go/src/github.com/apache/trafficcontrol
 VOLUME /root/go/src/github.com/apache/trafficcontrol
-ENV JPDA_OPTS="-agentlib:jdwp=transport=dt_socket,address=*:5005,server=y,suspend=n" JAVA_HOME=/usr/lib/jvm/java-11-openjdk M2_HOME=/root/go/src/github.com/apache/trafficcontrol/.m2 CATALINA_BASE=/opt/traffic_router
+ENV JPDA_OPTS="-agentlib:jdwp=transport=dt_socket,address=*:5005,server=y,suspend=n" JAVA_HOME=/usr/lib/jvm/java-11-openjdk M2_HOME=/root/go/src/github.com/apache/trafficcontrol/.m2 CATALINA_BASE=/opt/traffic_router TRAFFIC_MONITOR_HOSTS=trafficmonitor
 EXPOSE 3053:53/tcp 3053:53/udp 3080:80 3443:443 3333:3333 2222:3443 5005:5005
 
-RUN apk add --no-cache openjdk11 inotify-tools maven && ln -s /usr/lib/jvm/java-11-openjdk/bin/jdb /bin/jdb
+RUN apk add --no-cache openjdk11 inotify-tools maven tomcat-native openssl && ln -s /usr/lib/jvm/java-11-openjdk/bin/jdb /bin/jdb
 
 ADD https://archive.apache.org/dist/tomcat/tomcat-9/v9.0.43/bin/apache-tomcat-9.0.43.tar.gz /opt/tomcat.tgz
+
+ARG USER
+RUN adduser -H -D ${USER}
+RUN chown -R ${USER} /opt /usr/share/java/ /root
+USER ${USER}
```

Comment:
Oh wow, I didn't know setting `USER` and not `UID` would fix permissions

Reply:
Oh right, it's using the first available user ID, `1000`. So permissions are still not correct on macOS or other cases where your UID is not 1000

Reply:
I did it that way to avoid collisions in the UID from `adduser`. I think I just totally forgot that usernames are essentially syntactic sugar and that this won't mean anything. I'll try to get it working based on UID - but also be aware that not all shell/system combinations set the environment variable `UID` (or `USER` for that matter) and idk what to do about those systems.

Reply:
> but also be aware that not all shell/system combinations set the environment variable `UID` (or `USER` for that matter) and idk what to do about those systems.
AFAIK Windows doesn't care what user Docker says files are owned by

Reply:
Oh it's not just Windows. I actually first noticed it using `sh` in Alpine containers. They don't have `UID` or `USER` set in the execution environment, although `id -u` outputs correctly. It's not a problem with `sh`, either. `sh` in CentOS containers has it set. Using `zsh` on Pop!_OS seems to set `USER` in the environment, but only makes `UID` available as a shell variable, not actually exported to the environment. It seems to just be a personal choice by the OS whether or not to set those by default in new shell sessions.